subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/ProjectReportTaskIntegrationTest.groovy

```diff
@@ -107,4 +110,187 @@ Root project 'my-root-project' - this is a long description...
+            package com.example.restricted;
+
+            import org.gradle.api.Plugin;
+            import org.gradle.api.initialization.Settings;
+            import org.gradle.plugin.software.internal.SoftwareTypeRegistry;
+            import ${ RegistersSoftwareTypes.class.name};
+
+            @RegistersSoftwareTypes({ LibraryPlugin.class, ApplicationPlugin.class, UtilityPlugin.class })
+            abstract public class SoftwareTypeRegistrationPlugin implements Plugin<Settings> {
+                @Override
+                public void apply(Settings target) {}
+            }
+        """
+        file("build-logic/build.gradle") << """
+            plugins {
+                id("java-gradle-plugin")
+            }
+
+            ${mavenCentralRepository()}
+
+            gradlePlugin {
+                plugins {
+                    create("softwareTypeRegistrator") {
+                        id = "com.example.restricted.ecosystem"
+                        implementationClass = "com.example.restricted.SoftwareTypeRegistrationPlugin"
+                    }
+                }
+            }
+        """
+
+        and: "a build that applies that ecosystem plugin to a multi-project build, with each project using a different software type"
+        settingsFile << """
+            pluginManagement {
+                includeBuild("build-logic")
+            }
+
+            plugins {
+                id("com.example.restricted.ecosystem")
+            }
+
+            rootProject.name = 'example'
+
+            include("lib")
+            include("app")
+            include("util")
+        """
+        buildFile << """
+            project(":lib") {
+                description = "Sample library project"
+            }
+            project(":util") {
+                description = "Utilities and common code"
+            }
+            project(":app") {
+                description = "Sample application project"
+            }
+        """
+
+        file("lib/build.gradle.dcl") << """
+            library {
+                name = "my-lib"
+            }
+        """
+        file("app/build.gradle.dcl") << """
+            application {
+                name = "my-app"
+            }
+        """
+        file("util/build.gradle.dcl") << """
+            utility {
+                name = "my-util"
+            }
+        """
+
+        expect:
+        succeeds("projects")
+
+        outputContains("""
+------------------------------------------------------------
+Root project 'example'
+------------------------------------------------------------
+
+Root project 'example'
++--- Project ':app' - Sample application project
+        Software type: application (com.example.restricted.ApplicationExtension)
+        Defined in plugin: com.example.restricted.ApplicationPlugin
+        Registered by plugin: com.example.restricted.SoftwareTypeRegistrationPlugin
```

Comment:
From a UX POV, I think the plugin type here isn't as interesting or useful as the plugin id, but I don't think we keep track of that in a way that makes it easy to know. 

I think this is only an OK place to put this information, but a weird place for this to be permanently.  There are other things that might be interesting to show that's software definition related:

- The full set of available software definitions
- The location of conventions 
- The model as it's declared (a view of the project file + declarative conventions)
- The full model as it's realized

Maybe those last two belong to a project-specific task (like `model` does today).

In larger or nested projects, I think this also gets hard to read if the same software types are used over and over.

WDYT about...
- List all available software types and where they're defined before listing any projects
- Only include the name of the software type in the actual project summary

That would turn this into something like:
```
Available software types:

library (com.example.restricted.LibraryExtension)
        Defined in: com.example.restricted.LibraryPlugin
        Registered by: com.example.restricted.SoftwareTypeRegistrationPlugin

... (other software types)

Root project 'example'
+--- Project ':app' (application) - Sample application project
+--- Project ':lib' (library) - Sample library project
\\--- Project ':util' (utility) - Utilities and common code
```

Reply:
+1 to the idea of listing software type details separate from the projects.